src/hooks/usePublisher.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useRef, useState } from "react";
```

Comment:
If there is no UI code in hooks, we could use 'ts' instead of 'tsx' in file name

Reply:
Learnt something new here. THanks :)